src/libtls/tls_server.c

```diff
@@ -482,6 +484,27 @@ static status_t process_client_hello(private_tls_server_t *this,
 			this->client_version = version;
 		}
 	}
+
+	/* downgrade protection (see RFC 8446, section 4.1.3) */
+	if (original_max_version == TLS_1_3 &&
+		version < TLS_1_3)
+	{
+		char *downgrade_protection;
+
+		downgrade_protection = tls_downgrade_protection_tls11.ptr;
```

Comment:
To me it seems to be the easiest way to work with a pointer to char since the server-random value is stored as char anyway.